### PR TITLE
Scrolling & Panning added to Timeline

### PR DESCRIPTION
**Pull request by marcusnovotny**
Implemented basic scrolling & panning behaviour in the new d3 js layout

### CODE REVIEW

File: public/js/main.js [hunk 1]
```diff
@@ -85,22 +143,30 @@ function characterChart(svg, dataURL) {
...
+        d3.selectAll(".react").call(d3.behavior.zoom()
+            .x(x)
+            .scaleExtent([0.1, 10]) // TODO: Set meaningful values
+            .on("zoom", recalc));
+
         render();
     }
 
     function render() {
-        var s = getSize();
```
Comment (julienschmidt) [inline public/js/main.js:92]: micro-caching, yo!

**Comment by julienschmidt** [comment]
Can you maybe limit the scrolling to a range we actually have data for?

otherwise :+1:

**Comment by julienschmidt** [comment]
And give your pull request more meaningful titles :stuck_out_tongue_closed_eyes: (You can edit it here on Github)

**Comment by marcusnovotny** [comment]
Yeah I could, but I didn't want to limit my testing abilities for now. I'd leave it in and see how it works with actual data, ok?

**Comment by julienschmidt** [comment]
101692-mkay

**Pull request merged**